src/core/tsi/s2a/handshaker/s2a_handshaker_client.h

```diff
@@ -111,16 +111,47 @@ class S2AHandshakerClient {
    *  no further handshaker requests will be scheduled with the S2A. **/
   void Shutdown();
 
-  /** This method is exposed for testing purposes only. **/
-  grpc_byte_buffer* get_send_buffer_for_testing();
+  void MaybeCompleteTsiNext(
+      bool receive_status_finished,
+      s2a_recv_message_result* pending_recv_message_result);
+
+  /** When the refcount (given by |refs_|) of this S2A handshaker client
+   *  instance is positive, this method dereferences all of the data owned by
+   *  this S2A handshaker client instance. Otherwise, this method does
+   *  nothing. **/
+  void Unref();
+
+  /** These methods are exposed for use in the |on_status_received| callback
+   *  function. **/
+  grpc_status_code handshake_status_code() { return handshake_status_code_; }
+  const grpc_slice handshake_status_details() {
+    return handshake_status_details_;
+  }
+
+  /** These methods are exposed for testing purposes only. If |is_test_| is
+   *  false, then these methods do nothing or return nullptr. **/
```

Comment:
`and`?

Reply:
Sorry, I meant: if the return type is `void`, do nothing; if the return type is a pointer, then do nothing and return `nullptr`. I've clarified this comment in the next commit.